src/sentry/auth/helper.py

```diff
@@ -391,6 +391,15 @@ def _get_display_name(self, identity):
     def _get_identifier(self, identity):
         return identity.get('email') or identity.get('id')
 
+    def _find_existing_user(self, email):
+        return User.objects.filter(
+            id__in=UserEmail.objects.filter(
+                email__iexact=email,
+                is_verified=True,
```

Comment:
this is a slight change in behavior in that:

1. it doesnt prioritize username
2. it only allows merge with verified emails (this is actually something we should have done immediately from a security POV)